components/tooltip/demo/demo.html

```diff
@@ -0,0 +1,75 @@
+<!--Copyright (c) Coherent Labs AD. All rights reserved. Licensed under the MIT License. See License.txt in the project root for license information. -->
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="../coherent-gameface-components-theme.css">
+    <link rel="stylesheet" href="../style.css">
+    <title>Document</title>
+    <style>
+        .target {
+            width: 150px;
+            text-align: center;
+            background-color: #6e6d6d;
+            color: #ffffff;
+            position: relative;
+            top: 500px;
+            left: 500px;
+            padding: 10px;
+        }
+
+        .container {
+            display: flex;
+            flex-direction: column;
+        }
+
+        .title,
+        .button {
+            height: 40px;
+            background-color: var(--default-color-gray);
+            display: flex;
+            justify-content: center;
+            align-items: center;
+        }
+
+        .button {
+            background-color: var(--default-color-gray);
+            cursor: pointer;
+        }
+    </style>
+</head>
+
+<body>
+    <div class="target">Hover over me</div>
+    <gameface-tooltip id="tutorial" for=".target" on="click" position="top" off="click">
+        <div slot="message">
+            <div class="container">
+                <div class="title">Wellcome!</div>
+                <div class="text">
+                    Hello, this is your first tutorial.
+                    Click the next button to continue.
+                </div>
+                <div class="button">Next</div>
+            </div>
+        </div>
+    </gameface-tooltip>
+    <gameface-tooltip for=".target" on="mouseenter" position="left" off="mouseleave">
+        <div slot="message">Message on left</div>
+    </gameface-tooltip>
+    <gameface-tooltip for=".target" on="mouseenter" position="bottom" off="mouseleave">
+        <div slot="message">Message on bottom</div>
+    </gameface-tooltip>
+    <gameface-tooltip for=".target" on="mouseenter" position="right" off="mouseleave">
+        <div slot="message">Message on right</div>
+    </gameface-tooltip>
+    <script src="./bundle.js"></script>
+    <script>
+        document.getElementById('tutorial').addEventListener('click', (e) => {
+            e.currentTarget.hide();
+        });

```

Comment:
Isn't this supposed to be handled by the `off="click"` on the element?
If this is another use case, can we create a separate example?

Reply:
The tooltip can handle only events on itself. In this case we want to close it in click of a button inside it. It will be messy to enable this functionality through attributes because we will have to add another - offTarget or something that will make it confusing. This is why we manually add a listener to an element inside the tooltip and use the tooltip's hide method. This gives more flexibility, as it makes it possible to close the tooltip even outside of it from a different element.
What do you mean a separate example? It already is - the tooltip says "Click the next button to continue." and if you click it it will be closed.

Reply:
Everything you said is right! The concerns here are that the listener is added to the tooltip with id `tutorial` and not to the button that says `click the next button to continue`. You can change it to 
```
 document.querySelector('.button').addEventListener('click', (e) => {
         document.querySelector('#tutorial').hide();
 });
```
Also here we were confused if you wanted to add example with the `addEventListener` method or with the `off` attribute to close the tutorial and that is why we asked if you can split the examples :D

Reply:
I understand now. I really missed that the target is the whole tooltip, not only the component.
Fixed and added additional example.